### DISCUSSION TITLE
Integrating Web3Auth with current userbase

### DISCUSSION THREAD

**Discussion by ThianHooi** [Web3Auth/Web3Auth#869] [Help]
Our platform currently have a userbase and there is only one authentication method which is by email and password only. 

What are the best approach in implementing Web3Auth on our platform without users creating new account?

**Comment by shahbaz17** [comment]
Hey @ThianHooi 

You can bring your existing user base to Web3Auth. Checkout https://web3auth.io/docs/custom-authentication/byo-jwt-providers to see how you can do that by using a JWT-based authentication and passing that JWT to Web3Auth would enable you to onboard your users to Web3.

**Reply by ThianHooi** [reply]
@shahbaz17  Thanks for the reply. 

I need some help in implementation.

```typescript
const adapter = new OpenloginAdapter({
  adapterSettings: {
    network: 'testnet',
    clientId,
    uxMode: 'popup', // other option: popup
    loginConfig: {
      jwt: {
        name: 'lorem',
        verifier: 'lorem-testnet',
        typeOfLogin: 'jwt',
        // clientId: 'YOUR-CLIENTID-FROM-LOGIN-PROVIDER', <- [Q1] where to get this?
      },
    },
  },
});
```
In declaring the adapter, I am not sure where to get the `clientId` (marked as [Q1]). FYI, we are not using OAuth service. Our authentication is just simple passport-jwt

**Reply by shahbaz17** [reply]
Hey @ThianHooi 

When using a custom JWT, the `clientId` can be anything since this is a mandatory field by the SDK. So, simply put the `web3auth`'s `clientId`.

**Reply by ThianHooi** [reply]
@shahbaz17  I hope you don't mind to help me confirm the flow to `bring your existing user base to Web3Auth`. 

1. When existing users come to our platform, when they login with their email and password, API generates the JWT token
2. We then use the JWT token to authenticate with web3Auth to generate (or retrieve) private key for the wallet. Refer to [Q2] in the snippet below
```typescript
const web3authProvider = await web3auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
  loginProvider: "jwt",
  extraLoginOptions: {
    id_token: "idToken", // [Q2]: Insert API generated JWT token here?
    verifierIdField: "sub", // same as your JWT Verifier ID
  },
});
```

**Reply by ThianHooi** [reply]
@shahbaz17 Since web3auth is going to verify the JWT using the JWKS provided at the verifier. So I assume we will have to generate the JWT using the same JWKS?

**Reply by shahbaz17** [reply]
Yes, that's true. Please use the public-private key pair, and use the private key to encode it and expose the public key to an endpoint in a JWKS format.

Please look at https://web3auth.io/docs/custom-authentication/byo-jwt-providers#using-rsa-for-jwt-signing

**Reply by shahbaz17** [reply]
> 
> 1. When existing users come to our platform, when they login with their email and password, API generates the JWT token
>> Yes, that's true.
> 2. We then use the JWT token to authenticate with web3Auth to generate (or retrieve) private key for the wallet. Refer to [Q2] in the snippet below
> 
> ```ts
> const web3authProvider = await web3auth.connectTo(WALLET_ADAPTERS.OPENLOGIN, {
>   loginProvider: "jwt",
>   extraLoginOptions: {
>     id_token: "idToken", // [Q2]: Insert API generated JWT token here?
>     verifierIdField: "sub", // same as your JWT Verifier ID
>   },
> });
> ```
>> Yes, correct.
